source/lib/course-search/urls.ts

```diff
@@ -19,13 +20,48 @@ export let timeData = (options?: Options): Promise<string[]> =>
 	client.get('data-lists/valid_times.json', options).json()
 export let coursesForTerm = async (
 	term: TermType,
+	levels: Array<CourseType['level']>,
+	gereqs: string[] = [],
 	options?: Options,
 ): Promise<Array<CourseType>> => {
 	let data = (await client
 		.get(`${term.path}`, options)
 		.json()) as RawCourseType[]
-	return data.map((course) => ({
-		spaceAvailable: course.enrolled < course.max,
-		...course,
-	})) as CourseType[]
+	return data
+		.map((course) => ({
+			spaceAvailable: course.enrolled < course.max,
+			...course,
+		}))
+		.filter((c) => {
+			return findMatches(c.level, levels, c.gereqs, gereqs)
+		}) as CourseType[]
+}
+
+const findMatches = (
+	findLevel: number,
+	levels: Array<number>,
+	findgereqs: Array<string> = [],
+	gereqs: Array<string>,
+) => {
+	let foundLevels = matchesLevels(findLevel, levels)
```

Comment:
I kinda want to ask that we only run the checks that someone asked for?

Reply:
Great catch